src/Step/Deploy/InstallUpdate/Install/Setup/InstallCommandFactory.php

```diff
@@ -85,30 +107,36 @@ public function __construct(
         $this->stageConfig = $stageConfig;
         $this->elasticSuite = $elasticSuite;
         $this->dbConfig = $dbConfig;
+        $this->magentoVersion = $magentoVersion;
+        $this->elasticSearch = $elasticSearch;
     }
 
     /**
-     * Creates magento install command according to configured variables
-     *
      * @return string
+     * @throws ConfigException
      */
     public function create(): string
     {
-        $command = $this->getBaseCommand();
+        $command = 'php ./bin/magento setup:install';
 
-        /**
-         * Hack to prevent ElasticSuite from throwing exception.
-         */
-        if ($this->elasticSuite->isAvailable()) {
-            $host = $this->elasticSuite->get()['es_client']['servers'] ?? null;
+        if ($this->stageConfig->get(DeployInterface::VAR_VERBOSE_COMMANDS)) {
+            $command .= ' ' . $this->stageConfig->get(DeployInterface::VAR_VERBOSE_COMMANDS);
+        }
 
-            if ($host) {
-                $command .= ' --es-hosts=' . escapeshellarg($host);
-            }
+        try {
+            $options = array_replace(
+                $this->getBaseOptions(),
+                $this->getAdminOptions(),
+                $this->getEsOptions()
+            );
+        } catch (GenericException $exception) {
+            throw new ConfigException($exception->getMessage(), $exception->getCode(), $exception);
         }
 
-        if ($this->stageConfig->get(DeployInterface::VAR_VERBOSE_COMMANDS)) {
-            $command .= ' ' . $this->stageConfig->get(DeployInterface::VAR_VERBOSE_COMMANDS);
+        foreach ($options as $option => $value) {
+            $command .= ($value === null)
+                ? (' ' . $option)
```

Comment:
You have dupliation `'  ' . $option` in both variants, it can be moved before ternary operator, or changed to `sprintf` 
```
$command .=  sprintf(' %s%s', $option, $value === null ? '' : ' ' .  escapeshellarg($value)) 
```

Reply:
Thanks, looks good